esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -1142,6 +1192,34 @@ bool CWsESDLConfigEx::onGetESDLBinding(IEspContext &context, IEspGetESDLBindingR
     return true;
 }
 
+void CWsESDLConfigEx::addPublishHistory(IPropertyTree * publishedEntryTree, IEspPublishHistory & history)
+{
+    if (publishedEntryTree)
+    {
+         history.setLastEditBy(publishedEntryTree->queryProp("@lastEditedBy"));
+         history.setCreatedTime(publishedEntryTree->queryProp("@created"));
+         history.setPublishBy(publishedEntryTree->queryProp("@publishedBy"));
+         history.setLastEditTime(publishedEntryTree->queryProp("@lastEdit"));
+    }
+    else
+        ESPLOG(LogMin, "Could not fetch ESDL publish history!");
+}
+
+bool CWsESDLConfigEx::addESDLBindingContentsHistory(IPropertyTree * publishedEntryTree, IEspESDLBindingContents& esdlbindingcontents)
+{
+    //IEspESDLBindingContents extends IEspESDLBinding which contains the publish history for the binding
+    IEspESDLBinding * esdlbinding = dynamic_cast<IEspESDLBinding *>(&esdlbindingcontents);
```

Comment:
dynamic_cast might not be necessary here because IEspESDLBindingContents inherits from IEspESDLBinding.